### PR TITLE
provider/google: Implemented SSL LB caching agent.

### PR DESCRIPTION
**Pull request by jtk54**
Also makes some compatibility changes in other LB caching agents. @duftler @danielpeach please review.

### CODE REVIEW

**Comment by jtk54** [comment]
Ping -- ready to be reviewed, and github isn't down anymore :key:

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/agent/GoogleHttpLoadBalancerCachingAgent.groovy [hunk 1]
```diff
@@ -200,24 +201,24 @@ class GoogleHttpLoadBalancerCachingAgent extends AbstractGoogleCachingAgent impl
             portRange: forwardingRule.portRange,
             healths: [],
             hostRules: [],
-        )
-        loadBalancers << newLoadBalancer
+          )
+          loadBalancers << newLoadBalancer
+
```
Comment (duftler) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/agent/GoogleHttpLoadBalancerCachingAgent.groovy]: Extraneous blank line.

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/agent/GoogleSslLoadBalancerCachingAgent.groovy [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+
+    // Pass through objects
+    BatchRequest backendServiceRequest
+    BatchRequest healthCheckRequest
+    BatchRequest groupHealthRequest
+
+    @Override
+    void onSuccess(ForwardingRuleList forwardingRuleList, HttpHeaders responseHeaders) throws IOException {
+      forwardingRuleList?.items?.each { ForwardingRule forwardingRule ->
+        if (forwardingRule.target && Utils.getTargetProxyType(forwardingRule.target) == 'targetSslProxies') {
```
Comment (duftler) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/agent/GoogleSslLoadBalancerCachingAgent.groovy]: Any reason not to make `Utils.getTargetProxyType` return an `enum` type?

Comment (jtk54) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/agent/GoogleSslLoadBalancerCachingAgent.groovy]: Not particularly, I'll include that in a separate PR to keep changes minimal.

**Pull request merged**